Library/Homebrew/formulary.rb

```diff
@@ -222,7 +227,21 @@ def self.loader_for(ref)
       return FromPathLoader.new(ref)
     end
 
-    formula_with_that_name = Formula.path(ref)
+    if ref.start_with?(HOMEBREW_CELLAR) && File.directory?(ref)
+      rack = Pathname.new(ref)
+      name = rack.basename.to_s
+      kegs = rack.subdirs.map { |d| Keg.new(d) }
+      raise FormulaUnavailableError.new(name) if kegs.empty?
+      keg = kegs.detect(&:linked?) || kegs.detect(&:optlinked?) || kegs.max_by(&:version)
+      tap = Tab.for_keg(keg).tap
```

Comment:
I have just updated to make loading from rack an internal logic of Formulary. And under current context, it will definite not be a formula in core as TapFormulaAmbiguityError is only for taps.